src/app/modules/item/http-services/search-item.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { appConfig } from '../../../shared/helpers/config';
+import { ItemType } from '../../../shared/helpers/item-type';
+
+export interface ItemFound {
+  id: string,
+  title: string,
+  type: ItemType
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SearchItemService {
+
+  constructor(private http: HttpClient) {
+  }
+
+  search(
+    searchString: string,
+    includedTypes?: ItemType[],
+    excludedTypes?: ItemType[],
+    limit = 4,
+  ): Observable<ItemFound[]> {
+    let params = new HttpParams().set('search', searchString);
+
+    if (limit) params = params.set('limit', limit.toString());
```

Comment:
`limit` is always set. With `if (limit) ` you are only filtering "0" but there is no reason not to put the limit if =0.

Reply:
Fixed.